### PR TITLE
Technical challenge 1 Front end

### PR DESCRIPTION
**Pull request by Mar1789**
Added a page where users can input certain criteria to match with the best mentor

Test cases: 

Tested inputting matching criteria of a mentor to see if the correct mentor will output 
Screenshot 2024-07-11 at 12 36 46 PM

Tested each input option for ages, state, and interests to find any bugs.
Screenshot 2024-07-11 at 12 38 53 PM

Made sure each mentor are not duplicated as outputs when testing previous test cases
Screenshot 2024-07-11 at 12 39 42 PM

### CODE REVIEW

**Comment by Mar1789** [comment]
Note: This is the last pull request made before receiving feedback on my previous pull requests. Since these commits were made much earlier, I cannot make changes to split it up. Moving forward I will be able to apply the feedback I received.

File: Front-end/src/MatchMentor.jsx [hunk 1]
```diff
@@ -0,0 +1,244 @@
...
+      },
+    }).then((data) =>
+      data.json().then((data) => {
+        localStorage.removeItem("accessToken");
+        localStorage.removeItem("refreshToken");
+        window.location.href = "/";
+      })
+    );
+  }
+  async function member2() {
```
Comment (alisahq) [inline Front-end/src/MatchMentor.jsx:81]: Give it a better name? member2 is not evident on what it meant

File: Front-end/src/MatchMentor.jsx [hunk 2]
```diff
@@ -0,0 +1,244 @@
...
+        "Content-Type": "Application/json",
+      },
+    }).then((data) =>
+      data.json().then((data) => {
+        setInfo(data);
+      })
+    );
+  }
+  function match(e) {
+    let users = [];
```
Comment (alisahq) [inline Front-end/src/MatchMentor.jsx:94]: unused variable?

File: Front-end/src/components/Post.jsx [hunk 3]
```diff
@@ -171,7 +171,7 @@ const Post = (props) => {
           </p>
         </div>
         <h1 className="title">{props.title}</h1>
-        <h3 className="description">
+        <h3 className="p-description">
```
Comment (alisahq) [inline Front-end/src/components/Post.jsx:174]: QQ - what's the difference?

File: Front-end/src/MatchMentor.jsx (hunk 2 above)
Comment (Mar1789) [reply Front-end/src/MatchMentor.jsx:94]: It is being used in line 124 as an onsubmit

File: Front-end/src/components/Post.jsx (hunk 3 above)
Comment (Mar1789) [reply Front-end/src/components/Post.jsx:174]: p-description is to specify it was for posts. I had a css conflict with another element that had the same class name as "description"

File: Front-end/src/MatchMentor.jsx (hunk 1 above)
Comment (Mar1789) [reply Front-end/src/MatchMentor.jsx:81]: Updated

**Pull request merged**